### ISSUE TITLE
Lnurl auth problem with latest Spring Boot

### ISSUE THREAD

**Issue by straumat** [theborakompanioni/bitcoin-spring-boot-starter#105]
**Is your feature request related to a problem? Please describe.**

I use the latest Spring boot release and I try to implement lnurl auth.

Using your [example](https://github.com/theborakompanioni/bitcoin-spring-boot-starter/blob/a86e0fe31ae8d2d98a952539df88c94019d00cdb/incubator/spring-lnurl/spring-lnurl-auth-example-application/src/main/java/org/tbk/lightning/lnurl/example/LnurlAuthExampleApplicationSecurityConfig.java#L108), I get this error:
`'apply(C)' is deprecated since version 6.2 and marked for removal `

With latest spring boot release, `<C extends SecurityConfigurerAdapter<O, B>> C apply(C configurer)` is deprecated.
Now, the method `<C extends SecurityConfigurer<O, B>> C apply(C configurer)` should be used.

**Describe the solution you'd like**

Upgrade to latest spring boot release or tell me how I can make it work :)

**Comment by theborakompanioni** [comment]
Hey @straumat!

Thank you so much for the report. :muscle: 

In my local spring verison the deprecation text says `For removal in 7.0. Use {@link #with(SecurityConfigurerAdapter, Customizer)} instead.`.  Thank you for making me aware of it. See #106.

Please correct me if I am wrong but this means it is in your own application, right?
Is it sufficient to just apply the changes from the branch?
```patch
- apply(new LnurlAuthConfigurer()
+ with(new LnurlAuthConfigurer(), it -> it
```

Please let me know if I understood it correctly :pray: 




On side note, but unrelated:
The latest `devel` branch should be compatible with Spring Boot v3.2.1 and Spring v6.1.2. 
Since [OSSRH is down](https://status.maven.org/incidents/97rqnfx21p9n) (last check 2024-01-18) I cannot deploy a snapshot release at the moment. As soon as it is up again, I update `0.12.0-SNAPSHOT` and soon after `0.12.0`.

**Comment by straumat** [comment]
Thank you so much for the amazing, fast, and accurate support!

Indeed, with your solution `+ with(new LnurlAuthConfigurer(), it -> it`, the code compile. I did not finish yet to implement lnurl-auth in my application. I still have hard time understing why we have to keep k1 and not generate one each time the user log! I hope to write an article about how to implement it with your library. Thanks a lot for this work.

The work is in progress on my [Royllo project](https://github.com/royllo/explorer/tree/461-user-asset-data-management), it's an explorer for Taproot Assets. In fact, i'm not too at ease with this new way to code in Java, that's why I did not get the error.

Thanks again!
